extensions-contrib/distinctcount/src/main/java/io/druid/query/aggregation/distinctcount/DistinctCountAggregator.java

```diff
@@ -22,6 +22,7 @@
 import io.druid.collections.bitmap.MutableBitmap;
 import io.druid.query.aggregation.Aggregator;
 import io.druid.segment.DimensionSelector;
+import io.druid.segment.NullHandlingHelper;
 import io.druid.segment.data.IndexedInts;
 
 public class DistinctCountAggregator implements Aggregator
```

Comment:
DistinctCountAggregator seems to be broken, when selector doesn't have cardinality, or `idLookup()` returns null. E. g. when there is extraction function, SingleScanTimeDimSelector, etc.

Reply:
fixed.

Reply:
I don't know what have changed, but it is still broken, when there is extraction function. It will always return `IndexedInts` that look like [0], see all subclasses of `BaseSingleValueDimensionSelector`. It *does not* count distinct values.

Reply:
got it, looks like it is broken in current master also with BaseSingleValueDimensionSelector. 
and not a regression introduced in this PR,and its already a quite big PR, filed https://github.com/druid-io/druid/issues/5167